ibis/common/annotations.py

```diff
@@ -384,9 +435,10 @@ def annotated(_1=None, _2=None, _3=None, **kwargs):
 
     @functools.wraps(func)
     def wrapped(*args, **kwargs):
-        kwargs = sig.validate(*args, **kwargs)
-        result = sig.validate_return(func(**kwargs))
-        return result
+        values = sig.validate(*args, **kwargs)
```

Comment:
Add comments to describe the steps:

1. Argument validation
2. Call reconstruction
3. Actual function execution
4. Return value validation

Reply:
Done.